conda_smithy/feedstock_io.py

```diff
@@ -7,22 +7,28 @@
 def get_repo(path, search_parent_directories=True):
     repo = None
     try:
-        import git
+        import pygit2
 
-        repo = git.Repo(
-            path, search_parent_directories=search_parent_directories
-        )
+        if search_parent_directories:
+            path = pygit2.discover_repository(path)
+        if path is not None:
```

Comment:
What happens if `path is none`? Should we raise?

Reply:
The original behavior was to `return repo`, i.e. `None` then, and I've preserved that. In other words, if anything fails (`pygit2` is not installed, there is no repo, opening repo fails for some reason), the function returns `None`, as it used to.